### PR TITLE
Standardizing the Linting and Prettier Rules in App

### PR DESCRIPTION
**Pull request by jackblanc**
## Why

We need consistent eslint, tslint, and prettier rules throughout the applications we build. 

## This PR

Alters the configuration of eslint and tslint, adds prettier, and makes them all work together. 

Adds four commands to `package.json`:
```
npm run lint
npm run lint-fix
npm run prettier
npm run prettier-fix
```

This PR also adds Travis CI, which should run lint, prettier, and tests. 

## Next Steps

We need to decide if these rules work and if we all understand them. I don't personally 100% understand how this works.

### CODE REVIEW

**Comment by connernilsen** [comment]
@jackblanc can you add crlf/lf line endings to the ignore files? It's being all annoying on windows

**Comment by connernilsen** [comment]
Also getting this when I run prettier

```
$ npm run prettier

> frontend-scaffold@0.1.0 prettier C:\Users\conne\Documents\code\frontend-scaffold
> prettier --check './src/**/*.{js,jsx,ts,tsx,json,scss}'

Checking formatting...
[error] No files matching the pattern were found: "'./src/**/*.{js,jsx,ts,tsx,json,scss}'".
All matched files use Prettier code style!
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! frontend-scaffo
```

**Comment by jackblanc** [comment]
Yeah how do I add line endings? Not sure about that. 

The checks pass in CI, so I'm not sure why they wouldn't locally.

File: .eslintrc.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "prettier/prettier": [
       "error",
       {
-        "singleQuote": true
+        "singleQuote": true,
+        "endOfLine":"auto"
```
Comment (rymaju) [inline .eslintrc.json:20]: 🎉

File: package.json [hunk 2]
```diff
@@ -27,8 +27,11 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "lint": "eslint ./src/**/*.tsx",
-    "fix": "eslint ./src/**/*.tsx --fix"
+    "lint": "tslint -p . -c tslint.json && eslint ./src/**/*.tsx",
```
Comment (rymaju) [inline package.json:30]: Youre right `tslint` is more correct, great fix.

File: tslint.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "defaultSeverity": "error",
+  "extends": ["tslint:recommended", "tslint-config-prettier"],
+  "linterOptions": {
+    "exclude": [
+      "node_modules/*",
+      "src/serviceWorker.ts",
+      "src/setupTests.ts",
+      "src/react-app-env.d.ts"
```
Comment (rymaju) [inline tslint.json:9]: Do you want to add app.test here? I get a warning from that file when I lint.

**Review by rymaju — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (connernilsen) [reply package.json:30]: I'm getting this when I run it 

```
$ tslint -p . -c tslint.json && eslint ./src/**/*.tsx
Invalid option for configuration: tslint.json
```

File: .eslintrc.json (hunk 1 above)
Comment (connernilsen) [reply .eslintrc.json:20]: Do we want `"trailingComma": "all"` here?

File: package.json (hunk 2 above)
Comment (connernilsen) [reply package.json:30]: Wait it looks like it was because I didn't have a tslint.json file. Fixed now.

**Review by connernilsen — APPROVED** [review]
Looks good

File: .eslintrc.json (hunk 1 above)
Comment (jackblanc) [reply .eslintrc.json:20]: We can debate this to death, but I always prefer it, makes diffs super clean

File: package.json (hunk 2 above)
Comment (jackblanc) [reply package.json:30]: I am confused. Are you making local changes to this branch? That file should be included.

File: tslint.json (hunk 3 above)
Comment (jackblanc) [reply tslint.json:9]: I think it'll get deleted so no need

**Pull request merged**